api/controllers/DashboardController.js

```diff
@@ -133,177 +240,66 @@ var dashboardController = {
-
-                    if (err) {
-                        sails.log.error('Casebook Status Retrieval API error');
-                    } else if (api_results[0].length === 0) {
-                        sails.log.error('No Casebook Statuses available');
-                    }
-                    // Build the application reference status obj. This contains the application reference and it's status
-                    // as a key/value pair.
-
-                    let appRef = {};
-                    let trackRef = {};
-
-                    if (api_results[0]) {
-                        for (let result of api_results[0]) {
-                            appRef[result.applicationReference] = result.status;
-                            trackRef[result.applicationReference] =
-                                result.trackingReference;
-                        }
-                    }
-
-                    // For each element in the database results array, add the application reference status
-                    // if one exists.
-
-                    for (let result of results) {
-                        const uniqueAppId = result.unique_app_id;
-                        const appStatus = appRef.hasOwnProperty(uniqueAppId)
-                            ? appRef[uniqueAppId]
-                            : null;
-
-                        result.app_status =
-                            dashboardController._userFriendlyStatuses(
-                                appStatus,
-                                result.applicationtype
-                            );
-                        result.tracking_ref = trackRef.hasOwnProperty(uniqueAppId)
-                            ? trackRef[uniqueAppId]
-                            : null;
-                    }
-                }
+            for (let result of results) {
+                const uniqueAppId = result.unique_app_id;
+                const appStatus = appRef.hasOwnProperty(uniqueAppId)
+                    ? appRef[uniqueAppId]
+                    : null;
 
-                const pageAttributes = {
-                    message: req.flash('info'),
-                    users_applications: results,
-                    moment,
-                    offset,
-                    sortOrder,
-                    paginationMessage,
-                    currentPage,
-                    totalPages,
-                    searchCriteria,
-                    user_data: userData,
-                    application_total: totalApplications,
-                };
-
-                return dashboardController._redirectToPage(
-                    pageAttributes,
-                    req,
-                    res
+                result.app_status = dashboardController._userFriendlyStatuses(
+                    appStatus,
+                    result.applicationtype
                 );
+                result.tracking_ref = trackRef.hasOwnProperty(uniqueAppId)
```

Comment:
are you checking for `null` specifically? i.e. `if(result.tracking_ref === null)`? If not I think `result.tracking_ref = trackRef[uniqueAppId]` would suffice. It will give you undefined but it would take out one more logic check out (trackRef.hasOwnProperty).

Reply:
that error usually occurs when you try to access a property on an undefined value, however you've got `let trackRef = {}`. so `trackRef.anything` will not cause the app to crash, since `trackRef` is not an undefined value. If you tried to chain one more time, `trackRef.somethingUndefined.somethingElse` that will definitely cause an error. 

When you use the `&&` operator, you definitely need to make sure you are not doing a strict check for a null or undefined value with `===` somewhere else in the code. You will end up with either `false` or `trackRef[uniqueAppId]`.

Reply:
ah fair point, I forgot about the `trackRef = {}`, yep you're right, with that in place it should be fine. I'll adjust the code accordingly :)